pkg/rsstorage/servers/s3server/s3_enumeration.go

```diff
@@ -19,6 +19,7 @@ import (
 type AwsOps interface {
 	BucketDirs(bucket, s3Prefix string) ([]string, error)
 	BucketObjects(bucket, s3Prefix string, concurrency int, recursive bool, reg *regexp.Regexp) ([]string, error)
+	BucketObjectsMap(bucket, s3Prefix string, concurrency int, recursive bool, reg *regexp.Regexp) (map[string]string, error)
```

Comment:
Looks good, I'd just suggest making this more obvious that it's map of ETag values. That could be a comment, but maybe renaming the method works better. Or if it's possible we'd need more metadata, a map of structs that might someday contain LastModified time or whatever.